src/vote.rs

```diff
@@ -44,6 +44,11 @@ impl<T: Serialize + Clone> Vote<T> {
         })
     }
 
+    /// Compose a Vote.
+    pub fn compose(payload: T, signature: Signature) -> Self {
```

Comment:
This function allows you to create a `Vote` where the signature doesn't match the payload. I'd consider marking it `unsafe`, but not entirely sure about that. At the very least, do not expose it outside of the crate, so make it `pub(crate)` or something.